set.mm

```diff
@@ -541087,6 +541090,76 @@ coordinates of a barycenter of two points in one dimension (complex
+  Monoid of endomorphisms
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  $c End $.
+  $( Token for the monoid of endomorphisms. $)
+  cend $a class End $. $( Syntax for the monoid of endomorphisms. $)
+
+  ${
+    $d c x $.
+    $( The monoid of endomorphisms on an object of a category.  (Contributed by
+       BJ, 4-Apr-2024.) $)
+    df-bj-end $a |- End = ( c e. Cat |-> ( x e. ( Base ` c ) |->
+                 { <. ( Base ` ndx ) , ( x ( Hom ` c ) x ) >. ,
+                   <. ( +g ` ndx ) , ( <. x , x >. ( comp ` c ) x ) >. } ) ) $.
+  $}
+
+  ${
+    $d C c x $.  $d X c x $.  $d ph c x $.
+    bj-endval.c $e |- ( ph -> C e. Cat ) $.
+    bj-endval.x $e |- ( ph -> X e. ( Base ` C ) ) $.
+    $( Value of the monoid of endomorphisms on an object of a category.
+       (Contributed by BJ, 5-Apr-2024.) $)
+    bj-endval $p |- ( ph -> ( ( End ` C ) ` X ) =
+                   { <. ( Base ` ndx ) , ( X ( Hom ` C ) X ) >. ,
+                     <. ( +g ` ndx ) , ( <. X , X >. ( comp ` C ) X ) >. } ) $=
+      ( vx vc cnx cbs cfv cv chom co cop cco cpr cvv wceq fveq2 opeq2d cend a1i
+      cplusg cmpt ccat df-bj-end preq12d mpteq12dv wcel fvex fvmptd3 id oveq12d
+      oveqd mptex opeq12d adantl prex fvmptd ) AFCHIJZFKZVABLJZMZNZHUCJZVAVANZV
+      ABOJZMZNZPZUTCCVBMZNZVECCNZCVGMZNZPZBIJZBUAJQAGBFGKZIJZUTVAVAVRLJZMZNZVEV
+      FVAVROJZMZNZPZUDFVQVJUDZUEUAQFGUFVRBRZFVSWFVQVJVRBISWHWBVDWEVIWHWAVCUTWHV
+      TVBVAVAVRBLSUNTWHWDVHVEWHWCVGVFVAVRBOSUNTUGUHDWGQUIAFVQVJBIUJUOUBUKVACRZV
+      JVPRAWIVDVLVIVOWIVCVKUTWIVACVACVBWIULZWJUMTWIVHVNVEWIVFVMVACVGWIVACVACWJW
+      JUPWJUMTUGUQEVPQUIAVLVOURUBUS $.
+
+    $( Base set of the monoid of endomorphisms on an object of a category.
+       (Contributed by BJ, 5-Apr-2024.) $)
+    bj-endbase $p |- ( ph ->
+                      ( Base ` ( ( End ` C ) ` X ) ) = ( X ( Hom ` C ) X ) ) $=
+      ( cnx cbs cfv cend chom co cop cplusg cco cpr bj-endval fveq1d cvv fvexd
+      c1 df-base 1nn strndxid wne wceq basendxnplusgndx fvex ovex fvpr1 3eqtr3d
+      mp1i ) AFGHZCBIHZHZHULULCCBJHZKZLFMHZCCLCBNHKZLOZHZUNGHUPAULUNUSABCDEPQAU
+      NGTRACUMSUAUBUCULUQUDUTUPUEAUFULUQUPURFGUGCCUOUHUIUKUJ $.
+
+    $( Composition law of the monoid of endomorphisms on an object of a
+       category.  (Contributed by BJ, 5-Apr-2024.) $)
+    bj-endcomp $p |- ( ph ->
+             ( +g ` ( ( End ` C ) ` X ) ) = ( <. X , X >. ( comp ` C ) X ) ) $=
+      ( cnx cplusg cfv cend cbs chom co cop cco cpr bj-endval fveq1d cvv fvexd
+      c2 df-plusg 2nn strndxid wne wceq basendxnplusgndx fvex ovex mp1i 3eqtr3d
+      fvpr2 ) AFGHZCBIHZHZHULFJHZCCBKHLZMULCCMZCBNHZLZMOZHZUNGHUSAULUNUTABCDEPQ
+      AUNGTRACUMSUAUBUCUOULUDVAUSUEAUFUOULUPUSFGUGUQCURUHUKUIUJ $.
+
+    $d C c x y z $.  $d X c x y z $.  $d ph c x y z $.
+    $( The monoid of endomorphisms on an object of a category is a monoid.
+       (Contributed by BJ, 5-Apr-2024.) $)
+    bj-endmnd $p |- ( ph -> ( ( End ` C ) ` X ) e. Mnd ) $=
```

Comment:
Great! I didn't think this was so easy to prove, but maybe a lot of supporting theorems were already available. I will have a closer look at it after the changes are merged.